src/NuGet.Clients/NuGet.CommandLine/Commands/UpdateCommand.cs

```diff
@@ -131,6 +131,26 @@ public override async Task ExecuteCommandAsync()
             await UpdateAllPackages(solutionDir, context);
         }
 
+        private async Task UpdateSelfAsync()
+        {
+            PackageSource targetSource;
+            switch (Source.Count)
+            {
+                case 0:
+                    targetSource = new PackageSource(NuGetConstants.V3FeedUrl);
+                    break;
+                case 1:
+                    // Use the package source from the load config to preload any creds that might be needed for authentication.
+                    var availableSources = SourceProvider.LoadPackageSources().Where(source => source.IsEnabled).ToList();
```

Comment:
I don't like that this duplicates code from the Extension... and I'm not convinced with an initial look that we need to build the entire List in this layer. 
If duplicating, why not just duplicate the [ResolveSource logic](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/Utility/PackageSourceProviderExtensions.cs#L17)?

Something like this avoids loading N sources up front?...
```suggestion
                    var availableSources = SourceProvider.LoadPackageSources().FirstOrDefault(f =>f.Source.IsEnabled && (f.Source.Equals(source, StringComparison.OrdinalIgnoreCase) || f.Name.Equals(source, StringComparison.OrdinalIgnoreCase))).ToList();
```

Reply:
I removed the ToList so now it's happening lazily. 

Now I don't think we need to duplicate any code.